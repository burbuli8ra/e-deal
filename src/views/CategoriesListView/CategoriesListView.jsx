import React, { useEffect, useState } from 'react';
import fetchRoute from 'api';
import { CategoriesList } from 'components';
import { useAppContext } from 'provider';
import { withLoader } from 'hoc';

const CategoriesListWithLoader = withLoader(CategoriesList);

const CategoriesListView = () => {
  const [categories, setCategories] = useState([]);

  const [{ isPageLoading }, appDispatch] = useAppContext();

  const getCategoriesList = async () => {
    appDispatch({ type: 'SET_IS_LOADING' });
    const response = await fetchRoute.categoriesList();
    return response;
  }

  useEffect(() => {
    getCategoriesList()
      .then(response => {
        appDispatch({ type: 'RESET_IS_LOADING' });

        // Error handling
        if (response.hasOwnProperty('error')) {
          setCategories([]);
          appDispatch({
            type: 'SET_ERROR',
            payload: response.error
          });
          return;
        }

        // Order items by position
        const sortedResponse = response.sort((a, b) =>
          a.position - b.position
        );

        appDispatch({ type: 'RESET_ERROR' });
        setCategories(sortedResponse);
      })
  }, []);

  return (
    <CategoriesListWithLoader
      categories={categories}
      isLoading={isPageLoading}
    />
  );
};

export default CategoriesListView;