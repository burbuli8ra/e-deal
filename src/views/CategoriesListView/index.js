export { default as CategoriesListView } from './CategoriesListView';