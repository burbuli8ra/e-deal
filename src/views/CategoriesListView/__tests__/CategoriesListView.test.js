import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { act, render } from '@testing-library/react';
import fetchRoute from 'api';
import Mocks from '__mocks__';
import { AppProvider } from 'provider';
import CategoriesListView from '../CategoriesListView';

const setup = props => (
  <AppProvider>
    <Router>
      <CategoriesListView {...props} />
    </Router>
  </AppProvider>
);

describe('CategoriesListView component', () => {
  let props;
  let wrapper;

  const getCategories = jest.spyOn(fetchRoute, 'categoriesList')

  beforeEach(async () => {
    props = {};

    getCategories.mockImplementation(() => Mocks.CategoriesList);

    await act(async () => {
      wrapper = render(setup(props));
    });
  });

  test('should match snapshot', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  });
});