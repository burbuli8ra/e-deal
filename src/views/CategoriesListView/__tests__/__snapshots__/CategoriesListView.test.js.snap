// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesListView component should match snapshot 1`] = `
<DocumentFragment>
  <article
    class="css-1lu5s7y"
    data-testid="categoryItem-block"
    to="/categories/6989"
  >
    <a
      class="css-1ttiqzk"
      href="/categories/6989"
    >
      <img
        alt="Τεχνολογία"
        class="css-jd31iv"
        src="https://abpcdn.pstatic.gr/P/bpimg128/6989_SX400Y400/1577836800/technology.webp"
      />
      Τεχνολογία
    </a>
  </article>
  <article
    class="css-1lu5s7y"
    data-testid="categoryItem-block"
    to="/categories/2185"
  >
    <a
      class="css-1ttiqzk"
      href="/categories/2185"
    >
      <img
        alt="Σπίτι & Κήπος"
        class="css-jd31iv"
        src="https://abpcdn.pstatic.gr/P/bpimg128/2185_SX400Y400/1577836800/home-garden.webp"
      />
      Σπίτι & Κήπος
    </a>
  </article>
  <article
    class="css-1lu5s7y"
    data-testid="categoryItem-block"
    to="/categories/2068"
  >
    <a
      class="css-1ttiqzk"
      href="/categories/2068"
    >
      <img
        alt="Μόδα"
        class="css-jd31iv"
        src="https://abpcdn.pstatic.gr/P/bpimg128/2068_SX400Y400/1577836800/fashion.webp"
      />
      Μόδα
    </a>
  </article>
  <article
    class="css-1lu5s7y"
    data-testid="categoryItem-block"
    to="/categories/583"
  >
    <a
      class="css-1ttiqzk"
      href="/categories/583"
    >
      <img
        alt="Υγεία & Ομορφιά"
        class="css-jd31iv"
        src="https://abpcdn.pstatic.gr/P/bpimg128/583_SX400Y400/1577836800/health-beauty.webp"
      />
      Υγεία & Ομορφιά
    </a>
  </article>
  <article
    class="css-1lu5s7y"
    data-testid="categoryItem-block"
    to="/categories/2175"
  >
    <a
      class="css-1ttiqzk"
      href="/categories/2175"
    >
      <img
        alt="Παιδικά - Βρεφικά"
        class="css-jd31iv"
        src="https://abpcdn.pstatic.gr/P/bpimg128/2175_SX400Y400/1577836800/paidika-brefika.webp"
      />
      Παιδικά - Βρεφικά
    </a>
  </article>
  <article
    class="css-1lu5s7y"
    data-testid="categoryItem-block"
    to="/categories/3058"
  >
    <a
      class="css-1ttiqzk"
      href="/categories/3058"
    >
      <img
        alt="Αθλητισμός, Hobby"
        class="css-jd31iv"
        src="https://abpcdn.pstatic.gr/P/bpimg128/3058_SX400Y400/1577836800/sports-hobbies.webp"
      />
      Αθλητισμός, Hobby
    </a>
  </article>
  <article
    class="css-1lu5s7y"
    data-testid="categoryItem-block"
    to="/categories/3204"
  >
    <a
      class="css-1ttiqzk"
      href="/categories/3204"
    >
      <img
        alt="Μηχανοκίνηση"
        class="css-jd31iv"
        src="https://abpcdn.pstatic.gr/P/bpimg128/3204_SX400Y400/1577836800/auto-moto.webp"
      />
      Μηχανοκίνηση
    </a>
  </article>
  <article
    class="css-1lu5s7y"
    data-testid="categoryItem-block"
    to="/categories/6988"
  >
    <a
      class="css-1ttiqzk"
      href="/categories/6988"
    >
      <img
        alt="Ψυχαγωγία"
        class="css-jd31iv"
        src="https://abpcdn.pstatic.gr/P/bpimg128/6988_SX400Y400/1577836800/entertainment.webp"
      />
      Ψυχαγωγία
    </a>
  </article>
</DocumentFragment>
`;
